pkg/lib/crd/serialize.go

```diff
@@ -51,18 +51,64 @@ func Serialize(manifest string) (*apiextensionsv1.CustomResourceDefinition, erro
 	// set CRD type meta
 	// for purposes of fake client for unit tests to pass
 	crd.TypeMeta.Kind = Kind
-	crd.TypeMeta.APIVersion = APIVersion
-
+	crd.TypeMeta.APIVersion = Group + V1Version
 
 	// for each version in the CRD, check and make sure there is a schema
 	// if not a schema, give a default schema of props
 	for i := range crd.Spec.Versions {
 		if crd.Spec.Versions[i].Schema == nil {
 			schema := &apiextensionsv1.JSONSchemaProps{Type: "object"}
-			crd.Spec.Versions[i].Schema = &apiextensionsv1.CustomResourceValidation{OpenAPIV3Schema:schema}
+			crd.Spec.Versions[i].Schema = &apiextensionsv1.CustomResourceValidation{OpenAPIV3Schema: schema}
```

Comment:
the schema will prune everything. Is that intentional?

Reply:
Instead refuse these CRDs. They are wrong. Don't try to be clever to fix them up.